rclpy/src/rclpy/_rclpy_signal_handler.c

```diff
@@ -21,92 +21,182 @@
 
 #include <signal.h>
 
+#if __APPLE__ || _POSIX_C_SOURCE >= 1 || _XOPEN_SOURCE || _POSIX_SOURCE
```

Comment:
I took this verbatim from [rclcpp](https://github.com/ros2/rclcpp/blob/master/rclcpp/src/rclcpp/signal_handler.hpp#L35), so I'd be inclined to make a follow-up PR addressing this on both repositories. What do you think? Maybe @wjwwood can weigh in as he's the author of the piece of code I wrote this after.

Reply:
That might be a nice improvement. I usually tend to lean towards code that stands on its own, rather than depending on cmake to provide context, but the cmake check is likely more portable.

I took the said incantation from some SO most likely. All of the defines used are well known and documented in various places, so I don't think the above is incorrect. Essentially it translates to:

- apple for sure has it, because it's certified POSIX
  - https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/sigaction.2.html (link is for iPhoneOS, but it's still Darwin, I have that man page locally)
- Is the POSIX version >= 1 (sigaction was not available before that)
  - https://stackoverflow.com/a/20460042/671658
- Does it implement X/Open System Interface (XSI)? it also has sigaction
  - https://pubs.opengroup.org/onlinepubs/009604499/functions/sigaction.html
- `_POSIX_SOURCE` is defined in all recent versions of POSIX, usually something like `200112L`, and again, POSIX has sigaction
  - https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html

Reply:
Roger. I'm inclined to the define's as is then. If it doesn't work on some platform, someone using that platform will hopefully open an issue about it.